samtranslator/model/api/api_generator.py

```diff
@@ -304,6 +304,17 @@ def _construct_api_domain(self, rest_api):
 
         domain.EndpointConfiguration = {"Types": [endpoint]}
 
+        mutual_tls_auth = self.domain.get("MutualTlsAuthentication", None)
+        if mutual_tls_auth and isinstance(mutual_tls_auth, dict):
+            domain.MutualTlsAuthentication = {}
+            if mutual_tls_auth.get("TruststoreUri", None):
```

Comment:
We should add an exception if we get any keys that are not covered by these cases, to avoid failing open on a typo, for example.